File: subarr_sum.py
n_inputs = int(input())
for i in range(n_inputs):
    n, s = input().strip().split(" ")
    n = int(n)
    s = int(s)
    nums = [int(v) for v in input().strip().split(" ")]

    solution = -1
    for j in range(0, n):
        summ = 0
        for r in range(j, n):
            summ += nums[r]
            if summ == s:
                solution = "{} {}".format(j+1, r+1)
            if summ > s:
                break

        if solution != -1:
            break
    print(solution)

File: subarr_sum.cpp
#include <cstdio>

int main(){

    int n_inputs;
    scanf("%d", &n_inputs);

    for(int i=0; i< n_inputs; i++){

        int n, s;
        char space;
        scanf("%d %d", &n, &s);
        int nums [n];
        for(int j=0; j <n;j++)
            scanf("%d", &nums[j]);

        int i1 = -1;
        int i2 = 0;

        for(int j=0; j<n; j++){
            int sum = 0;
            for(int r=j; r<n; r++){
                sum += nums[r];
                if(sum == s){
                   i1 = j+1;
                   i2 = r+1;
                   break;
                }
                if(sum > s){
                    break;
                }
            }
            if(i1 != -1)
                break;
        }

        if(i1 != -1)
                printf("%d %d\n",i1, i2);
        else
            printf("%d\n", -1);

    }
    return 0;
}
